apis/v1beta1/gateway_types.go

```diff
@@ -522,6 +627,12 @@ type GatewayConditionType string
 // particular Gateway condition type has been raised.
 type GatewayConditionReason string
 
+const (
+	// This reason is used with "Programmed" and "Accepted" conditions when
+	// desired routability is not able to be fulfilled by the implementation
+	GatewayUnsupportedRoutability GatewayConditionReason = "UnsupportedRoutability"
+)
```

Comment:
Better to add this to the list of other conditions below.

Reply:
Done in https://github.com/kubernetes-sigs/gateway-api/pull/2157